doc/devel/api_changes.rst

```diff
@@ -137,6 +143,30 @@ Expire deprecation
      instead. For removed items that were not in the stub file, only deleting from the
      allowlist is required.
 
+.. _pending-deprecation:
+
+Pending deprecations
+^^^^^^^^^^^^^^^^^^^^
+
+If the deprecation is introduced as part of implementing an alternative, then the
+deprecation is marked as pending and introduced one meso release after the feature.
+
+For example, if the feature is released in 3.Y, then code that should be deprecated
+because of the feature is marked with the appropriate deprecation decorator and
+the following settings:
+* the *pending* parameter is set to ``True``
+* the *removal* parameter is left blank
+
+Once the version with the feature (e.g. 3.Y) has been released, the pending deprecation
+is introduced. The decorator should be updated such that:
+* *pending* is set to ``False``
+* *since* is set to the next meso release (3.Y+1)
+* *removal* is set to at least 2 versions after (3.Y+3) introduction.
```

Comment:
```suggestion
When switching from peding to an actual deprecation, the decorator should
be updated such that:
* *pending* is set to ``False``
* *since* is set to the next meso release (3.Y+1)
* *removal* is set to at least 2 versions after (3.Y+3) introduction.
```